pytorch_lightning/utilities/argparse.py

```diff
@@ -106,7 +107,7 @@ def parse_env_variables(cls, template: str = "PL_%(cls_name)s_%(cls_argument)s")
     return Namespace(**env_args)
 
 
-def get_init_arguments_and_types(cls) -> List[Tuple[str, Tuple, Any]]:
+def get_init_arguments_and_types(cls: Union[Type['pl.Trainer'], Any]) -> List[Tuple[str, Tuple, Any]]:
```

Comment:
just `Any` would be sufficient?

Reply:
but that way you get type hints for trainer.

Reply:
But this isn't right as the `Trainer` is not the only one to use these classes.

And the fact that they are public API means the users can use these without the `Trainer`.

so @stancld can you update it?